Make OSSF scorecard happy

trask opened the pull request:
This didn't work: #13164

Trying a more conventional setup.

(OSSF scorecard is supposed to look for a successful run of this and then not warn on binary gradle wrappers)

After merging, I'll update the required status checks.

breedx-splk reviewed (approved):
Referencing the new workflow is the follow-up work then right?

trask commented:
> Referencing the new workflow is the follow-up work then right?

add reference in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/.github/repository-settings.md? yeah, I added that in this PR now (and added the required check in the settings)

trask commented:
> I added that in this PR now (and added the required check in the settings)

nope, I got the name wrong, I think I need to merge this PR first, then I can add the new check as required in the rule set
